spec/API_specification/array_object.md

```diff
@@ -396,12 +396,16 @@ Converts a zero-dimensional boolean array to a Python `bool` object.
 
 
 (method-__dlpack__)=
-### \_\_dlpack\_\_(/, *, stream=None)
+### \_\_dlpack\_\_(x, /, *, stream=None)
```

Comment:
This change is incorrect. `__dlpack__` is a method which returns a capsule with the data of the array inside. It does not take any positional parameters.

Same comment for `__dlpack_device__`.

Reply:
The methods in this file include the `self` argument (except they are called `x` or `x1` instead of `self`, which I personally don't like, but that's the way it is). Unless this is supposed to be a static method, which isn't mentioned (if that's the case, why isn't it just a function?). But since it says "exports **the** array", I'm assuming that's not the case.

Reply:
This change is consistent with the rest of the array object specification.

Whether to document `self` was discussed previously. Originally, the spec approach was based on [NumPy](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.__abs__.html#numpy.ndarray.__abs__)'s and [TF](https://www.tensorflow.org/api_docs/python/tf/Tensor#methods_2)'s docs, which both documented `self`. Both were and still are inconsistent; NumPy less so, with now fewer dunder methods having documentation including `self` as a documented argument. Python [docs](https://docs.python.org/3/reference/datamodel.html#object.__lt__) do document self.

The use of `x` and `x#` over `self` and, e.g., `other`, was done following discussed spec guidance that recommended array argument names use consistent conventions across the spec. Functional APIs use `x` and `x#`, and that practice was continued for the array object methods.

However, not using `self` for the first argument of instance methods goes against [PEP 8](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments). Accordingly, I think we should adopt `self` and, where necessary, `other` for method names.

As this was my original doing, I can submit a PR that updates the array object method documentation.

Reply:
Agreed, we should either leave it out or use `self`, anything else will be very confusing.

Reply:
The `self` change was made, resolving.

Reply:
Almost merged, in gh-135